configure script should ensure presence of MPI

Right now if I build with autotools and forget to set CC=mpicc, configure succeeds, but the build fails with:
```
libtool: link: gcc -g -Wall -o .libs/test_async_multicomp test_async_multicomp.o test_common.o  -L/usr/local/pnetcdf-1.12.1/lib -L/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib ../../src/clib/.libs/libpioc.so /usr/local/pnetcdf-1.12.1/lib/libpnetcdf.so /usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib/libnetcdf.so -pthread -Wl,-rpath -Wl,/usr/local/pnetcdf-1.12.1/lib -Wl,-rpath -Wl,/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib
libtool: link: gcc -g -Wall -o .libs/test_decomp_uneven test_decomp_uneven.o test_common.o  -L/usr/local/pnetcdf-1.12.1/lib -L/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib ../../src/clib/.libs/libpioc.so /usr/local/pnetcdf-1.12.1/lib/libpnetcdf.so /usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib/libnetcdf.so -pthread -Wl,-rpath -Wl,/usr/local/pnetcdf-1.12.1/lib -Wl,-rpath -Wl,/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib
libtool: link: gcc -g -Wall -o .libs/test_darray_async_simple test_darray_async_simple.o test_common.o  -L/usr/local/pnetcdf-1.12.1/lib -L/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib ../../src/clib/.libs/libpioc.so /usr/local/pnetcdf-1.12.1/lib/libpnetcdf.so /usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib/libnetcdf.so -pthread -Wl,-rpath -Wl,/usr/local/pnetcdf-1.12.1/lib -Wl,-rpath -Wl,/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib
libtool: link: gcc -g -Wall -o .libs/test_darray_fill test_darray_fill.o test_common.o  -L/usr/local/pnetcdf-1.12.1/lib -L/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib ../../src/clib/.libs/libpioc.so /usr/local/pnetcdf-1.12.1/lib/libpnetcdf.so /usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib/libnetcdf.so -pthread -Wl,-rpath -Wl,/usr/local/pnetcdf-1.12.1/lib -Wl,-rpath -Wl,/usr/local/netcdf-c-4.7.4-development_hdf5-1.10.6_szip/lib
/usr/bin/ld: test_common.o: undefined reference to symbol 'MPI_Comm_split'
//usr/local/lib/libmpi.so.12: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
Makefile:691: recipe for target 'test_async_multicomp' failed
make[3]: *** [test_async_multicomp] Error 1
```

So that's not right. If we can't build without MPI we can say so in configure.

--- a/.github/workflows/a4.yml
+++ b/.github/workflows/a4.yml
@@ -128,18 +128,4 @@ jobs:
         autoreconf -i
         ./configure --enable-fortran
         make -j check
-    - name: cmake build
-      run: |
-        set -x
-        echo 'export PATH=/home/runner/mpich/bin:$PATH' > .bashrc
-        source .bashrc
-        export CC=mpicc
-        mkdir build
-        cd build
-        export LD_LIBRARY_PATH="/home/runner/netcdf-c/lib:/home/runner/pnetcdf/lib:/home/runner/hdf5/lib:/home/runner/mpich/lib:$LD_LIBRARY_PATH"
-        cmake -Wno-dev -DNetCDF_C_LIBRARY=/home/runner/netcdf-c/lib/libnetcdf.so -DNetCDF_C_INCLUDE_DIR=/home/runner/netcdf-c/include -DPnetCDF_PATH='/home/runner/pnetcdf' -DPIO_ENABLE_FORTRAN=Off -DPIO_HDF5_LOGGING=On -DPIO_USE_MALLOC=On -DPIO_ENABLE_LOGGING=On -DPIO_ENABLE_TIMING=Off .. || (cat CMakeFiles/CMakeOutput.log && cat CMakeFiles/CMakeError.log)
-        cat config.h
-        make VERBOSE=1
-        make tests VERBOSE=1
-        ctest -VV
         

--- a/configure.ac
+++ b/configure.ac
@@ -236,6 +236,12 @@ fi
 
 AC_MSG_NOTICE([finding libraries])
 
+# Ensure we have MPI.
+AC_CHECK_FUNCS([MPI_Init])
+if test "x$ac_cv_func_MPI_Init" != "xyes"; then
+   AC_MSG_ERROR([Can't link to MPI library. MPI is required.])
+fi
+
 # Check for netCDF library.
 AC_CHECK_LIB([netcdf], [nc_create], [], [AC_MSG_ERROR([Can't find or link to the netcdf library.])])
 AC_CHECK_HEADERS([netcdf.h netcdf_meta.h])
@@ -332,9 +338,6 @@ if test "x${ac_cv_type_MPI_Offset}" = xyes; then
 else
    AC_MSG_ERROR([Unable to find type MPI_Offset in mpi.h])
 fi
-
-#AC_CHECK_SIZEOF([MPI_Offset], [], [[#include <mpi.h>]])
-#AC_DEFINE([SIZEOF_MPI_OFFSET], [8], [netCDF classic library available])
 
 # If we want the timing library, we must find it.
 if test "x$enable_timing" = xyes; then

